### PR TITLE
special filtering for created/updated_at at midnight

### PR DESCRIPTION
**Pull request by ajtmccarty**
IFC-1947

- add new special handling that allows filtering for a whole day's worth of metadata if the input time is midnight in any timezone
  - basically, if you filter to Interfaces with `node_metadata__created_at=2025-12-31T00:00:00`, then this will be translated to `node_metadata__created_at__after=2025-12-30T23:59:59.999999` and `node_metadata__created_at__before=2026-01-01T00:00:00`
- updates to use class-scoped fixtures for some existing metadata filtering tests to make them run faster. probably a good pattern to use in other places

## Summary by CodeRabbit

* **Bug Fixes**
  * Metadata datetime filtering: exact-midnight created_at/updated_at values are now treated as full-day ranges when querying.

* **Behavior**
  * Day-range interpretation is applied automatically during query handling so midnight datetimes return that day's results.

* **Tests**
  * Added comprehensive unit tests for midnight/day-range behavior and many edge cases.
  * Improved test infrastructure with class-scoped fixtures and reusable schema helpers for better isolation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: backend/infrahub/graphql/resolvers/resolver.py [hunk 1]
```diff
@@ -167,6 +199,7 @@ async def default_paginated_list_resolver(
         filters = {
             key: value for key, value in kwargs.items() if ("__" in key and value is not None) or key in ("ids", "hfid")
         }
+        filters = _transform_metadata_day_filters(filters)
```
Comment (ajtmccarty) [inline backend/infrahub/graphql/resolvers/resolver.py:210]: I don't think we do this kind of filter transform anywhere else and we don't have a great system for it b/c filters are just dictionaries with string keys.
it would be better if this `filters` input was more structured and then this logic could be linked to that structure

File: backend/tests/conftest.py [hunk 2]
```diff
@@ -170,6 +180,13 @@ async def default_branch(reset_registry, local_storage_dir, empty_database, db:
     return await do_default_branch(db=db)
 
 
+@pytest.fixture(scope="class")
+async def default_branch_scope_class(
+    reset_registry_scope_class, local_storage_dir_scope_class, empty_database_scope_class, db: InfrahubDatabase
```
Comment (ogenstad) [inline backend/tests/conftest.py]: Can you add type annotations to these fixtures? It will soon be enforced based on other PRs and we could end up with failing pilelines if some branches are out of date before we merge them.

File: backend/tests/unit/conftest.py [hunk 3]
```diff
@@ -1723,11 +1732,24 @@ async def criticality_schema(
...
-    )
+    return do_criticality_schema(branch=default_branch, schema_root=criticality_schema_root)
+
+
+@pytest.fixture(scope="class")
+async def criticality_schema_scope_class(
+    db: InfrahubDatabase,
+    default_branch_scope_class: Branch,
+    group_schema_scope_class,
+    data_schema_scope_class,
```
Comment (ogenstad) [inline backend/tests/unit/conftest.py]: Same comment as above.

**Review by ogenstad — APPROVED** [review]

File: backend/tests/conftest.py (hunk 2 above)
Comment (ogenstad) [reply backend/tests/conftest.py]: FYI: #7985
